packages/kbn-plugin-helpers/src/tasks/write_server_files.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { pipeline } from 'stream';
+import { promisify } from 'util';
+
+import vfs from 'vinyl-fs';
+import { transformFileWithBabel, transformFileStream } from '@kbn/dev-utils';
+
+import { BuildContext } from '../build_context';
+
+const asyncPipeline = promisify(pipeline);
+
+export async function writeServerFiles({
+  log,
+  config,
+  plugin,
+  sourceDir,
+  buildDir,
+  kibanaVersion,
+}: BuildContext) {
+  log.info('copying source into the build and converting with babel');
+
+  // copy source files and apply some babel transformations in the process
+  await asyncPipeline(
+    vfs.src(
+      [
+        'kibana.json',
+        ...(plugin.manifest.server
+          ? config.serverSourcePatterns || [
+              'yarn.lock',
+              'tsconfig.json',
+              'package.json',
+              'index.{js,ts}',
+              '{lib,server,common,translations}/**/*',
+            ]
+          : []),
+      ],
+      {
+        cwd: sourceDir,
+        base: sourceDir,
+        buffer: true,
+        ignore: [
+          '**/*.d.ts',
+          '**/public/**',
```

Comment:
Might be causing the `public/assets` not being copied to the `build` folder (#83387 )

Reply:
Hmm, that makes sense. Will get a PR up for 7.10.1 Monday.

You can probably either add the public/assets directory to the `serverSourcePatterns` setting.

If you're only dealing with a couple files you could also just import the file using the file-loader, ie https://github.com/elastic/kibana/blob/b8e2e85578f31296efac58580b1d8a306b8874e8/x-pack/plugins/maps/public/connected_components/mb_map/mb_map.js#L19

Nice find, sorry about that.

Reply:
Glad to hear you know how to fix it 👍 

I've tried adding `public/**/*`, `public/assets/**/*`, etc in `serverSourcePatterns`, but nothing worked. My guess is that it's linked to the same issue.

I have a workaround for now: build the plugin with  `--skip-archive`, manually add the content of `public/assets` to the build folder, and manually package the zip file.

Example with plugin named `redelk`:
```
yarn build --skip-archive
mkdir build/kibana/redelk/public
cp -R public/assets/ build/kibana/redelk/public
cd build
zip redelk-7.10.0.zip kibana -r
```

Reply:
Thanks to share your workaround!